src/impl_haiku.cc

```diff
@@ -0,0 +1,46 @@
+#include "iana-time-zone/include/impl_haiku.h"
+#include "iana-time-zone/src/tz_haiku.rs.h"
+
+#include <cstring>
+#include <Errors.h>
+#include <LocaleRoster.h>
+#include <String.h>
+#include <TimeZone.h>
+
+size_t ::tz_haiku::get_tz(rust::Slice<uint8_t> buf) {
+    try {
+        static_assert(sizeof(char) == sizeof(uint8_t), "Illegal char size");
+
+        if (buf.empty()) {
+            return 0;
+        }
+
+        BLocaleRoster *locale_roster(BLocaleRoster::Default());
+        if (!locale_roster) {
+            return 0;
+        }
+
+        BTimeZone tz(NULL, NULL);
+        if (locale_roster->GetDefaultTimeZone(&tz) != B_OK) {
+            return 0;
+        }
+
+        BString bname(tz.ID());
+        int32_t length(bname.Length());
```

Comment:
I don't know if this actually compiles, but can we do something like this:

```suggestion
        size_t length(size_t(bname.Length()));
```

Reply:
ahh maybe it's not that simple since we check if `length < 0` just beyond this. This is a spot where I prefer Rust shadowing ability.

Reply:
I introduced a new variable after checking that it's not negative. Yes, rust's variable shadowing is a very nice feature that I guess would work well with C++'s scoping rules, too. But I don't think that C++ will ever adopt this feature. Otherwise code like this wouldn't compile anymore: https://godbolt.org/z/8s8zM9K6h

```cpp
int main(int, char**) {
    int result(result);
    return result;
}
```